woof

```diff
@@ -216,10 +226,15 @@ class FileServHTTPRequestHandler (http.server.BaseHTTPRequestHandler):
          self.send_header ("Content-Type", "text/html")
          self.send_header ("Content-Length", str (len (txt)))
          self.end_headers ()
+
+         if headers_only:
+            return
+
          self.wfile.write (txt)
          return
 
-      maxdownloads -= 1
+      if not headers_only:
```

Comment:
The counter logic is too complicated, I'm not sure of a practical use case when we want to decrease a count if HEAD is hit.

HEAD is very welcome, especially for links shared to clients running them automatically.

Reply:
Why should a HEAD request count towards a total number of downloads if it — by definition — does not download the file?

Reply:
I think I misread your comment. What's complicated about it? This patch _does_ prevent decrementing the download limit.

Reply:
Ah perfect if it only decrease the counter when it hits GET, regardless any configuration option :)